Sources/App/Core/PackageContributorService.swift

```diff
@@ -0,0 +1,133 @@
+    public let numberOfCommits: Int
+    public let email: String
+    public let name: String
+    
+}
+
+
+/// Loads the contributors history from a Git repository
+struct GitHistoryLoader {
+    
+    static func loadContributorsHistory(cacheDirPath: String, packageID: UUID?) throws -> [Contributor] {
+        do {
+            let commitHistory = try queryVCHistory(cacheDirPath: cacheDirPath, packageID: packageID)
+            return try parseVCHistory(log: commitHistory)
+        } catch {
+            throw AppError.analysisError(packageID, "loadContributorsHistory failed: \(error.localizedDescription)")
+        }
+    }
+    
+    /// Gets the version control history in a string log
+    private static func queryVCHistory(cacheDirPath: String, packageID: UUID?) throws -> String {
+        
+        if !Current.fileManager.fileExists(atPath: cacheDirPath) {
+            throw AppError.unexistentPackageCacheDir(packageID, cacheDirPath)
+        }
+
+        // attempt to shortlog
+        do {
+            return try Current.git.shortlog(cacheDirPath)
+        } catch {
+            throw AppError.shellCommandFailed("gitShortlog",
+                                              cacheDirPath,
+                                              "queryVCHistory failed: \(error.localizedDescription)")
+        }
+    }
+    
+    /// Parses the result of queryVCHistory into a collection of contributors
+    private static func parseVCHistory(log: String) throws -> [Contributor] {
+        var committers = [Contributor]()
+        
+        for line in log.components(separatedBy: .newlines) {
+            let log = line.split(whereSeparator: { $0 == " " || $0 == "\t"})
+            
+            if (log.count != 2) {
+                var identifier = [String]()
+                for i in 1..<(log.count - 1) {
+                    identifier.append(String(log[i]))
+                }
+                
+                let committer = Contributor(numberOfCommits: Int(log.first!) ?? 0,
+                                            email: String(log.last!),
+                                            name: identifier.joined(separator: " "))
+                committers.append(committer)
+            }
+            
+        }
+        return committers
+    }
+    
+}
+
+
+
+/// Strategy for selecting contributors based entirely on the number of commits 
+struct CommitSelector {
+    
+    static func filter(candidates: [Contributor], threshold: Float) -> [Contributor] {
+        if candidates.isEmpty {
+            return []
+        }
+        
+        let maxNumberOfCommits = candidates.max(by: { (a,b) -> Bool in
+            return a.numberOfCommits < b.numberOfCommits
+        })!.numberOfCommits
+        
+        return candidates.filter { canditate in
+            return Float(canditate.numberOfCommits) > threshold * Float(maxNumberOfCommits)
+        }
+    }
+}
+
+
+struct PackageAuthors : Encodable, Decodable, Equatable {
```

Comment:
Nit: We could potentially structure the order of these types to make more sense for others coming to look at this file later. I'd suggest we start with `PackageAuthors` being the first type in the file, followed by `PackageContributorService` and finally the `GitHistoryLoader`, which could even be nested inside `PackageContributorService`.